### ISSUE TITLE
Eliminate Microsoft.CodeAnalysis pre-built in source-build

### ISSUE THREAD

**Issue by oleksandr-didyk** [dotnet/roslyn-analyzers#6619]
Part of https://github.com/dotnet/roslyn-analyzers/issues/6606

`Microsoft.CodeAnalysis: 3.3.1; 3.7.0; 3.8.0; 3.9.0` are direct dependencies that are being flagged as a pre-built in source-build.

The versions are referenced by the following projects:
  - `3.3.1` -> referenced by `ReleaseNotesUtil`
  - `3.7.0` -> referenced by `GenerateDocumentationAndConfigFiles`
  - `3.8.0` -> referenced by `GenerateDocumentationAndConfigFilesForBrokenRuntime`
  - `3.9.0` -> referenced by `Microsoft.CodeAnalysis.ResxSourceGenerator` directly (i.e. using a hard-coded value instead of `Versions.props`)

Could we utilize the latest version for these projects? Tried doing this by directly bumping the versions, but encountered a build-time exception:
```
/src/Roslyn.Diagnostics.Analyzers/Core/AbstractRunIterations`1.cs(39,32): error CS8600: Converting null literal or possible null value to non-nullable type. [/src/Roslyn.Diagnostics.Analyzers/Core/Roslyn.Diagnostics.Analyzers.csproj]
/src/Roslyn.Diagnostics.Analyzers/Core/AbstractRunIterations`1.cs(40,18): error CS8604: Possible null reference argument for parameter 'method' in 'bool IMethodSymbolExtensions.IsBenchmarkOrXUnitTestMethod(IMethodSymbol method, ConcurrentDictionary<INamedTypeSymbol, bool> knownTestAttributes, INamedTypeSymbol? benchmarkAttribute, INamedTypeSymbol? xunitFactAttribute)'. [/src/Roslyn.Diagnostics.Analyzers/Core/Roslyn.Diagnostics.Analyzers.csproj]
/src/Roslyn.Diagnostics.Analyzers/Core/AbstractApplyTraitToClass`1.cs(83,41): error CS8602: Dereference of a possibly null reference. [/src/Roslyn.Diagnostics.Analyzers/Core/Roslyn.Diagnostics.Analyzers.csproj]
src/Roslyn.Diagnostics.Analyzers/Core/AbstractApplyTraitToClass`1.cs(103,51): error CS8602: Dereference of a possibly null reference. [/src/Roslyn.Diagnostics.Analyzers/Core/Roslyn.Diagnostics.Analyzers.csproj]
```
I am also currently investigating the possibility of using reference assemblies for some of these versions and will update this issue based on my progress.

Additionally:
  - it seems that `ReleaseNotesUtil`, the project using `Microsoft.CodeAnalysis` is utilized for generating an executable for tooling used during the release. Is this something that should be a part of a repo build (i.e. can we exclude it from source-build)?
  - if we cannot use the latest version, could we use the same minor version for each of the references?

**Comment by oleksandr-didyk** [comment]
This issue was resolved by https://github.com/dotnet/roslyn-analyzers/issues/6606, just wasn't mentioned in the PR.

Closing
